internal/k8shandler/logstore_lokistack.go

```diff
@@ -28,6 +37,35 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateLokiStackLogStore() e
 	return nil
 }
 
+func findDefaultLokiHost(outputs []logging.OutputSpec) (string, error) {
+	for _, o := range outputs {
+		if o.Name == logging.OutputNameDefault {
+			if o.Type != logging.OutputTypeLoki {
+				return "", fmt.Errorf("expected loki output, got: %v", o.Type)
+			}
+			u, err := url.Parse(o.URL)
+			if err != nil {
+				return "", err
+			}
+			return u.Host, nil
+		}
+	}
+	return "", errors.New("no default output")
+}
```

Comment:
I think this could be simplified by generating the hostname based on the LokiStack name and ClusterLogging namespace, similar to the code in [`forwarding.go`](https://github.com/openshift/cluster-logging-operator/blob/b09b736e639b24cf1e28b7391111726e7745e46c/internal/k8shandler/forwarding.go#L384):

```suggestion
func findDefaultLokiHost(clusterLogging *logging.ClusterLogging) (string, error) {
	logStore := clusterLogging.Spec.LogStore
	if logStore == nil || logStore.Type != logging.LogStoreTypeLokiStack {
		return "", fmt.Errorf("logstore does not use LokiStack: %v", logStore)
	}

	return fmt.Sprintf("%s-query-frontend-http.%s.svc", logStore.LokiStack.Name, clusterLogging.Namespace), nil
}
```

Reply:
Have a look at the latest.